contrib/rss_downloader/src/rss_downloader.nit

```diff
@@ -256,23 +263,40 @@ redef class Text
 	fun to_rss_elements: Array[Element]
 	do
 		var xml = to_xml
+		if xml isa XMLError then
+			print_error "RSS Parse Error: {xml.message}:{xml.location or else "null"}"
+			return new Array[Element]
+		end
 		var items = xml["rss"].first["channel"].first["item"]
 
 		var elements = new Array[Element]
 		for item in items do
 			var title = item[tool_config.tag_title].first.as(XMLStartTag).data
-			var link = item["link"].first.as(XMLStartTag).data
+			var link = item[tool_config.tag_link].first.as(XMLStartTag).data
 
 			elements.add new Element(title, link)
 		end
 
 		if sys.verbose then
 			print "# Found elements:"
-			print elements.join("\n")
+			print "* " + elements.join("\n* ")
 		end
 
 		return elements
 	end
+
+	# Expand the Lempel-Ziv encoded `self`
+	fun gunzip: String
+	do
+		var proc = new ProcessDuplex("gunzip", new Array[String]...)
+		proc.write self
+		proc.stream_out.close
+		var res = proc.read_all
+		proc.stream_in.close
+		proc.wait
```

Comment:
It seems I see this idiom to duplex pipe a process, feed it then get the results. Maybe libify it.
There is also a risk of deadlock here since if the process output while the reading in not finished, the result pipe might fill up. thus blocking the process, thus cannot read more, thus the read pipe fill up, thus blocking the nit program.